pulp_deb/app/viewsets/repository.py

```diff
@@ -22,7 +27,33 @@
 from pulp_deb.app import models, serializers, tasks
 
 
-class AptRepositoryViewSet(RepositoryViewSet, ModifyRepositoryActionMixin):
+class AptModifyRepositoryActionMixin(ModifyRepositoryActionMixin):
+    @extend_schema(
+        description="Trigger an asynchronous task to create a new repository version.",
+        summary="Modify Repository Content",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    @action(detail=True, methods=["post"], serializer_class=RepositoryAddRemoveContentSerializer)
+    def modify(self, request, pk):
+        remove_content_units = request.data.get("remove_content_units", [])
+        package_hrefs = [href for href in remove_content_units if "/packages/" in href]
+
+        if package_hrefs:
+            prc_hrefs = self._get_matching_prc_hrefs(package_hrefs)
+            remove_content_units.extend(prc_hrefs)
+            request.data["remove_content_units"] = remove_content_units
+
+        return super().modify(request, pk)
+
+    def _get_matching_prc_hrefs(self, package_hrefs):
+        package_ids = [extract_pk(href) for href in package_hrefs]
+        matching_packages = Package.objects.filter(pulp_id__in=package_ids)
+        matching_components = PackageReleaseComponent.objects.filter(package__in=matching_packages)
+        prc_href = [get_url(component) for component in matching_components]
+        return prc_href
```

Comment:
Naming nitpicks:
```suggestion
        matching_prcs = PackageReleaseComponent.objects.filter(package__in=matching_packages)
        prc_hrefs = [get_url(prc) for prc in matching_prcs]
        return prc_hrefs
```
Because the name `components` makes me think of the `ReleaseComponent` class, not the `PackageReleaseComponent` class.